### ISSUE TITLE
Automatically detect HTML notes in ENEX files and import as HTML

### ISSUE THREAD

**Issue by laurent22** [laurent22/joplin#9606]
### Operating system

Windows

### Joplin version

2.13.11

### Current behaviour

Notes that are created from the web clipper in Evernote are in special read-only HTML mode. And the user can edit only after converting them to an editable format.

We should detect these notes and automatically import them as HTML.

### Sample files

- Clip as "full page": [Full page.enex.zip](https://github.com/laurent22/joplin/files/13785979/Full.page.enex.zip)

- Clip as "article": [Article.enex.zip](https://github.com/laurent22/joplin/files/13785981/Article.enex.zip)

**Comment by varunkumar-22** [comment]
I can see the issue still exists when Evernote file is imported as markdown.

For the provided `Article.enex` file, importing it as markdown results in a completely empty note. Importing the same file as HTML works initially and the content appears correctly.however when clicking “Convert to Markdown” joplin creates a new empty markdown note and moves the original HTML note to trash. When opening the note from trash and attempting the conversion again the following error appears: “Could not convert notes to Markdown: Cannot convert read-only item: Article”. 

I think this is not just about detecting the HTML notes properly but more of HTML to markdown conversion failure. This could potentially lead to data loss if unnoticed

**Note**: full page file worked good for both HTML and markdown import 
Operating System: MacOs

<img width="1512" height="982" alt="Image" src="https://github.com/user-attachments/assets/6d9b8229-d05b-4a12-8daa-0174148e00e9" />

**Comment by varunkumar-22** [comment]
@laurent22  I’m experiencing this on macOS, whereas the original issue mentions Windows. Since the environment is different, would you prefer that I open a separate issue to track this specifically for macOS?

**Comment by Guni102** [comment]
Hi! I’ve submitted a pull request that attempts to address this issue.

The PR updates the ENEX import logic so that notes containing HTML content are detected and stored with the correct markup_language instead of being incorrectly treated as Markdown.

Pull request: #14741

Please let me know if any changes or improvements are needed.
